packages/plugin-react/src/index.ts

```diff
@@ -50,14 +50,20 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
   let isProduction = true
   let skipFastRefresh = opts.fastRefresh === false
   let filter = createFilter(opts.include, opts.exclude)
+  let jsxInject: string | undefined
 
   const userPlugins = opts.babel?.plugins || []
   const userParserPlugins =
     opts.parserPlugins || opts.babel?.parserOpts?.plugins || []
 
+  const importReactRegex = /(^|\n)import\s+(\*\s+as\s+)?React\s+/
+
   const viteBabel: Plugin = {
     name: 'vite:react-babel',
     enforce: 'pre',
+    configResolved(config) {
+      jsxInject = config.esbuild ? config.esbuild.jsxInject : undefined
```

Comment:
Better use `config.esbuild?.jsxInject` here

Reply:
`config.esbuild` can also be `false`. using `config.esbuild?.jsxInject` is a type error